sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java

```diff
@@ -255,27 +256,19 @@ public Resource build() {
     }
 
     private Resource.ReservationInfo getRefinedReservationInfo(String role, String resId) {
-        return Resource.ReservationInfo.newBuilder()
-                .setPrincipal(principal.get())
+        Resource.ReservationInfo.Builder reservationBuilder = Resource.ReservationInfo.newBuilder()
                 .setRole(role)
                 .setType(Resource.ReservationInfo.Type.DYNAMIC)
-                .setLabels(
-                        Protos.Labels.newBuilder()
-                                .addLabels(Protos.Label.newBuilder()
-                                        .setKey(MesosResource.RESOURCE_ID_KEY)
-                                        .setValue(resId)))
-                .build();
+                .setPrincipal(principal.get());
+        AuxLabelAccess.setResourceId(reservationBuilder, resId);
```

Comment:
I wonder if things like this are actually more appropriately placed next to the resources they modify; `Label` isn't really a meaningful cross-cutting concern in our world, instead we have type-local concerns like `resource_id`. Isolating the fact that it's even implemented as a label under the hood seems like the logical endpoint of abstracting label access.